src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        objects = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= objects.length) {
+            this.grow();
```

Comment:
```suggestion
            grow();
```

why do we need `this` keyword here?